src/main/scala/org/zalando/spearheads/innkeeper/dao/PathsPostgresRepo.scala

```diff
@@ -76,38 +76,67 @@ class PathsPostgresRepo @Inject() (
   override def collisionExistsForPath(path: PathIn): Future[Boolean] = {
     logger.debug("path uri collision check")
 
-    def position = SimpleExpression.binary[String, String, Int] {
+    val query = Paths
+      .filter(filterPathsWithUriCollision(path.uri, path.hasStar.contains(true)))
+      .map(_.hostIds)
+
+    db.run {
+      query.result
+    }.map{ existingPathHostIds =>
+      hostIdCollisionExists(path.hostIds, existingPathHostIds)
+    }
+  }
+
+  override def collisionExistsForPatch(pathId: Long, newHostIds: Seq[Long]): Future[Boolean] = {
```

Comment:
The existence of a method named `collisionExistsForPath` AND a method named `collisionExistsForPatch`, that operates on paths confuses the hell out of me. Also, I don't think `patch` is a word that should be used on this level. On the Postgres level, it is more like `collisionExistsForUpdate`, isn't it?